src/engine/Operation.cpp

```diff
@@ -69,17 +69,38 @@ void Operation::recursivelySetTimeConstraint(
   });
 }
 
+// _____________________________________________________________________________
+void Operation::updateRuntimeStats(bool applyToFilter, uint64_t numRows,
+                                   uint64_t numCols,
+                                   std::chrono::milliseconds duration) const {
+  auto& rti = applyToFilter || !externalFilterApplied_
+                  ? runtimeInfo()
+                  : *runtimeInfo().children_.at(0);
+  rti.totalTime_ += duration;
+  rti.originalTotalTime_ = rti.totalTime_;
+  rti.originalOperationTime_ = rti.getOperationTime();
+  if (!applyToFilter || externalFilterApplied_) {
+    rti.numRows_ += numRows;
+    rti.numCols_ = numCols;
+  }
+  if (!applyToFilter && externalFilterApplied_) {
```

Comment:
This is the exact negation of a condition from above, it could be given a speaking name.